main.cpp

```diff
@@ -104,5 +106,45 @@ int main()
         }
     }
 
+
+
+    // ------ test code --------
+
+    Hashtable nameTable(1000);
+    Hashtable tokenTabel(1000);
+
```

Comment:
Vorschlag:  Vielleicht noch eine Art Containerklasse bauen die die beiden Hashtables speichert und den Zugriff über string key, string keyType ermöglicht. Auf jedenfall würd ich den Code in eine  Testklasse auslagern und Methodennamen vergeben die genau saugen was du testest. Ansich eine sehr gute Idee, das Thema [Unit Testing](https://en.wikipedia.org/wiki/Unit_testing) könnte dich eventuell auch interessieren, vielleicht findest du ja eine einfache Möglichkeit für c++ sowas aufzusetzen, wenn genug Zeit da ist.